packages/infra/lib/hl7-notification-stack/mllp.ts

```diff
@@ -47,14 +49,25 @@ export class MllpStack extends cdk.NestedStack {
       "Allow outbound traffic from MLLP server"
     );
 
-    const nlb = new elbv2.NetworkLoadBalancer(this, "MllpServerNLB", {
+    const nlb = new elbv2.NetworkLoadBalancer(this, "MllpServerNLB2", {
       vpc,
       internetFacing: false,
-      vpcSubnets: {
-        subnets: vpc.privateSubnets,
-      },
     });
 
+    // Set static IP address for NLB in private subnet
+    const privateSubnet = vpc.privateSubnets[0];
+    if (!privateSubnet || vpc.privateSubnets.length !== 1) {
+      throw new Error("Should have exactly one private subnet");
+    }
+    const cfnNlb = nlb.node.defaultChild as elbv2.CfnLoadBalancer;
+    cfnNlb.subnets = undefined;
+    cfnNlb.subnetMappings = [
+      {
+        subnetId: privateSubnet.subnetId,
+        privateIPv4Address: MLLP_SERVER_NLB_INTERNAL_IP,
+      },
+    ];
```

Comment:
This whole thing is a hack as it's not possible to set a static internal ip using the high level NetworkLoadBalancer CDK construct, and it automatically sets subnets to the default subnets in the vpc if you don't specify anything.